xarray/core/missing.py

```diff
@@ -176,6 +176,15 @@ def get_clean_interp_index(arr, dim, use_coordinate=True, **kwargs):
         # raise if index cannot be cast to a float (e.g. MultiIndex)
         try:
             index = index.values.astype(np.float64)
+            if method != 'nearest':
+                # rescale index to avoid overflow/underflow
+                # The division can change the nearest-neighbour
+                # when compared to pandas (which does not divide).
+                # Let's keep that compatitibility
+                index = (index - index.min())
+                if len(index) > 1:
+                    index /= index.std()
```

Comment:
Standard deviation is *probably* fine here, but I wonder if it would be better to *guarantee* that are all values fall within some fixed range [-MAX, MAX], e.g., by using max() instead:
```
index = index - index.mean()
index /= EPSILON * np.maximum(index.max(), 1)
```
where EPSILON is some constant, e.g., 1e-8.